### PR TITLE
podman-remote image tree

### PR DESCRIPTION
**Pull request by baude**
add the ability for the podman-remote client to be able to print an
image tree.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: cmd/podman/tree.go [hunk 1]
```diff
@@ -1,23 +1,13 @@
...
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
-const (
-	middleItem   = "├── "
-	continueItem = "│   "
-	lastItem     = "└── "
-)
-
```
Comment (jwhonce) [inline cmd/podman/tree.go]: I think all this formatting should stay in the presentation level not be pushed into the library.

You can retrieve the same data structures that will be pushed over varlink and then format and present them. 

Unless some other command needs this formatting?

Comment (mheon) [reply cmd/podman/tree.go]: Agree, this should never be used by other commands

Comment (baude) [reply cmd/podman/tree.go]: where exactly do you want it so both local and remote clients can use it? and adapter is not considered "the library" remember?

Comment (mheon) [reply cmd/podman/tree.go]: Code can still live in cmd/podman - the output logic should be exactly the same, once we get the right structs transferred

**Comment by baude** [comment]
rebased and print moved

**Comment by jwhonce** [comment]
LGTM

File: cmd/podman/tree.go [hunk 2]
```diff
@@ -3,11 +3,11 @@ package main
 import (
 	"context"
 	"fmt"
+	"github.com/containers/libpod/libpod/image"
```
Comment (rhatdan) [inline cmd/podman/tree.go]: Wrong place

File: pkg/adapter/images.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+func (r *LocalRuntime) Tree(c *cliconfig.TreeValues) (*image.InfoImage, map[string]*image.LayerInfo, *ContainerImage, error) {
+	img, err := r.NewImageFromLocal(c.InputArgs[0])
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	// Fetch map of image-layers, which is used for printing output.
+	layerInfoMap, err := image.GetLayersMapWithImageInfo(r.Runtime.ImageRuntime())
+	if err != nil {
+		return nil, nil, nil, errors.Wrapf(err, "error while retriving layers of image %q", img.InputName)
```
Comment (rhatdan) [inline pkg/adapter/images.go]: retriving is spelled wrong.

File: pkg/adapter/images_remote.go [hunk 4]
```diff
@@ -5,6 +5,9 @@ package adapter
 import (
 	"context"
 	"encoding/json"
+	"github.com/containers/libpod/cmd/podman/cliconfig"
```
Comment (rhatdan) [inline pkg/adapter/images_remote.go]: Wrong place

File: pkg/adapter/images_remote.go [hunk 5]
```diff
@@ -22,3 +25,32 @@ func (i *ContainerImage) Inspect(ctx context.Context) (*inspect.ImageData, error
...
+	if err := json.Unmarshal([]byte(reply), &layerInfoMap); err != nil {
+		return nil, nil, nil, err
+	}
+
+	reply, err = iopodman.BuildImageHierarchyMap().Call(r.Conn, c.InputArgs[0])
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	if err := json.Unmarshal([]byte(reply), imageInfo); err != nil {
+		return nil, nil, nil, err
```
Comment (rhatdan) [inline pkg/adapter/images_remote.go]: Can you wrap this error.

File: pkg/adapter/images_remote.go [hunk 6]
```diff
@@ -22,3 +25,32 @@ func (i *ContainerImage) Inspect(ctx context.Context) (*inspect.ImageData, error
...
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "failed to obtain image layers")
+	}
+	if err := json.Unmarshal([]byte(reply), &layerInfoMap); err != nil {
+		return nil, nil, nil, err
+	}
+
+	reply, err = iopodman.BuildImageHierarchyMap().Call(r.Conn, c.InputArgs[0])
+	if err != nil {
+		return nil, nil, nil, err
```
Comment (rhatdan) [inline pkg/adapter/images_remote.go]: Does this error need a wrap.

**Comment by baude** [comment]
addressed @rhatdan 's comments

**Pull request merged**